spec/system/admin/enterprise_fees_spec.rb

```diff
@@ -191,28 +207,60 @@
       login_as enterprise_user
     end
 
-    it "creates enterprise fees" do
-      visit edit_admin_enterprise_path(distributor1)
-      within(".side_menu") { click_link 'Enterprise Fees' }
-      click_link "Create One Now"
-
-      select distributor1.name,
-             from: 'sets_enterprise_fee_set_collection_attributes_0_enterprise_id'
-      select 'Packing', from: 'sets_enterprise_fee_set_collection_attributes_0_fee_type'
-      fill_in 'sets_enterprise_fee_set_collection_attributes_0_name', with: 'foo'
-      select 'GST', from: 'sets_enterprise_fee_set_collection_attributes_0_tax_category_id'
-      select 'Flat Percent', from: 'sets_enterprise_fee_set_collection_attributes_0_calculator_type'
-      click_button 'Update'
-
-      expect(flash_message).to eq('Your enterprise fees have been updated.')
-
-      # After saving, we should be redirected to the fees for our chosen enterprise
-      expect(page)
-        .not_to have_select 'sets_enterprise_fee_set_collection_attributes_1_enterprise_id',
-                            selected: 'Second Distributor'
-
-      enterprise_fee = EnterpriseFee.find_by name: 'foo'
-      expect(enterprise_fee.enterprise).to eq(distributor1)
+    context "creating an enterprise fee" do
+      before do
+        visit edit_admin_enterprise_path(distributor1)
+        within(".side_menu") { click_link 'Enterprise Fees' }
+        click_link "Create One Now"
+      end
+
+      shared_examples "shared example" do |tax_category, calculator, flash_message, fee_count|
```

Comment:
> "shared example"

It would be better to have a meaningful name for this. Perhaps something like "setting tax category with a calculator". Hmm I see the duplication with the below context name, but still I think that's better than nothing.

Reply:
Nice catch, this one slipped. I think I've left the naming to the end, but probably forgot about it... 

Before it read:

```
As an administrator
    I want to manage enterprise fees
  as an enterprise manager
    creating an enterprise fee
      an error message is displayed
        behaves like shared example
          setting it up as Inherit From Product, with a Flat Rate (per order) calculator
            triggers the expected message
      an success message is displayed
        behaves like shared example
          setting it up as Inherit From Product, with a Flat Rate (per item) calculator
            triggers the expected message
      an success message is displayed
        behaves like shared example
          setting it up as GST, with a Flat Rate (per order) calculator
            triggers the expected message
      an success message is displayed
        behaves like shared example
          setting it up as GST, with a Flat Rate (per item) calculator
            triggers the expected message

```

After the latest commit, it reads:
```

As an administrator
    I want to manage enterprise fees
  as an enterprise manager
    creating an enterprise fee
      an error message is displayed
        behaves like setting it up
          as Inherit From Product, with a Flat Rate (per order) calculator
            triggers the expected message
      an success message is displayed
        behaves like setting it up
          as Inherit From Product, with a Flat Rate (per item) calculator
            triggers the expected message
      an success message is displayed
        behaves like setting it up
          as GST, with a Flat Rate (per order) calculator
            triggers the expected message
      an success message is displayed
        behaves like setting it up
          as GST, with a Flat Rate (per item) calculator
            triggers the expected message
```
Hope this is ok, let me know what you think.

Reply:
This makes sense 👍